pallets/sto/src/lib.rs

```diff
@@ -0,0 +1,398 @@
+        // Interacting with a fundraiser before the start `Moment`.
+        FundraiserNotStarted,
+        // Using an invalid venue
+        InvalidVenue,
+        // Using an invalid portfolio
+        InvalidPortfolio,
+        // An individual price tier was invalid or a set of price tiers was invalid
+        InvalidPriceTiers,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as StoCapped {
+        /// All fundraisers that are currently running. (ticker, fundraiser_id) -> Fundraiser
+        Fundraisers get(fn fundraisers): double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) u64 => Option<Fundraiser<T::Balance, T::Moment>>;
+        /// Total fundraisers created for a token
+        FundraiserCount get(fn fundraiser_count): map hasher(twox_64_concat) Ticker => u64;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Create a new offering. A fixed amount of pre-minted tokens are put up for sale at the specified tiered rate.
+        #[weight = 800_000_000]
+        pub fn create_fundraiser(
+            origin,
+            offering_portfolio: PortfolioId,
+            offering_asset: Ticker,
+            raising_portfolio: PortfolioId,
+            raising_asset: Ticker,
+            tiers: Vec<PriceTier<T::Balance>>,
+            venue_id: u64,
+            start: Option<T::Moment>,
+            end: Option<T::Moment>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(T::Asset::primary_issuance_agent(&offering_asset) == did, Error::<T>::Unauthorized);
+
+            let venue = VenueInfo::get(venue_id).ok_or(Error::<T>::InvalidVenue)?;
+            ensure!(
+                venue.creator == did && venue.venue_type == VenueType::Sto,
+                Error::<T>::InvalidVenue
+            );
+
+            ensure!(raising_portfolio.did == did, Error::<T>::InvalidPortfolio);
+            ensure!(<PortfolioAssetBalances<T>>::contains_key(raising_portfolio, raising_asset), Error::<T>::InvalidPortfolio);
+
+            ensure!(offering_portfolio.did == did, Error::<T>::InvalidPortfolio);
+            ensure!(<PortfolioAssetBalances<T>>::contains_key(offering_portfolio, offering_asset), Error::<T>::InvalidPortfolio);
+            let asset_balance = <PortfolioAssetBalances<T>>::get(offering_portfolio, offering_asset);
+
+            ensure!(
+                tiers.len() > 0 && tiers.iter().all(|t| t.total > 0.into()),
+                Error::<T>::InvalidPriceTiers
+            );
+
+            let offering_amount: T::Balance = tiers
+                .iter()
+                .map(|t| t.total)
+                .fold(0.into(), |total, x| total + x);
+            ensure!(offering_amount <= asset_balance, Error::<T>::InsufficientTokensRemaining);
```

Comment:
We should be calling `Portfolio::lock_tokens` to lock these tokens. That function does the amount check on its own so we don't need it here. 

The amount check being done here is not right anyway since the same tokens may already be committed to other settlements.